addons/kube-ovn/kube-ovn.yaml

```diff
@@ -0,0 +1,568 @@
+          command: ["/kube-ovn/install-cni.sh"]
+          securityContext:
+            runAsUser: 0
+            privileged: true
+          volumeMounts:
+            - mountPath: /opt/cni/bin
+              name: cni-bin
+      containers:
+        - name: cni-server
+          image: "kubeovn/kube-ovn:v1.10.0"
+          imagePullPolicy: IfNotPresent
+          command:
+            - bash
+            - /kube-ovn/start-cniserver.sh
+          args:
+            - --enable-mirror=false
+            - --encap-checksum=true
+            - --service-cluster-ip-range=10.152.183.0/24
+            - --iface=
+            - --network-type=geneve
+            - --default-interface-name=
+          securityContext:
+            runAsUser: 0
+            privileged: true
+          env:
+            - name: ENABLE_SSL
+              value: "false"
+            - name: POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          volumeMounts:
+            - mountPath: /etc/openvswitch
+              name: systemid
+            - mountPath: /etc/cni/net.d
+              name: cni-conf
+            - mountPath: /run/openvswitch
+              name: host-run-ovs
+            - mountPath: /run/ovn
+              name: host-run-ovn
+            - mountPath: /var/run/netns
+              name: host-ns
+              mountPropagation: HostToContainer
+            - mountPath: /var/log/kube-ovn
+              name: kube-ovn-log
+            - mountPath: /var/log/openvswitch
+              name: host-log-ovs
+            - mountPath: /var/log/ovn
+              name: host-log-ovn
+            - mountPath: /etc/localtime
+              name: localtime
+          livenessProbe:
+            failureThreshold: 3
+            initialDelaySeconds: 30
+            periodSeconds: 7
+            successThreshold: 1
+            tcpSocket:
+              port: 10665
+            timeoutSeconds: 3
+          readinessProbe:
+            failureThreshold: 3
+            initialDelaySeconds: 30
+            periodSeconds: 7
+            successThreshold: 1
+            tcpSocket:
+              port: 10665
+            timeoutSeconds: 3
+          resources:
+            requests:
+              cpu: 200m
+              memory: 200Mi
+            limits:
+              cpu: 1000m
+              memory: 1Gi
+      nodeSelector:
+        kubernetes.io/os: "linux"
+      volumes:
+        - name: systemid
+          hostPath:
+            path: /var/snap/microk8s/common/etc/origin/openvswitch
+        - name: host-run-ovs
+          hostPath:
+            path: /run/openvswitch
+        - name: host-run-ovn
+          hostPath:
+            path: /run/ovn
+        - name: cni-conf
+          hostPath:
+            path: /var/snap/microk8s/current/args/cni-network
```

Comment:
Here we mount in the pod the `/var/..../current/....`. As current is a symlink this translates to `/var/.../<revision>/....`. As soon as the snap refreshes the mounted directory on the host (`/var/.../<revision>/...`) is invalid as there is now a new revision. So after a couple of revisions the snap will not work. This problem in not new, the same issue exists in calico. For this reason when we refresh the snap we redeploy calico [1]. We should be doing something similar for kube-ovn.

A slightly different issue that we need to address is how refreshes happen in multi-node clusters. When a node refreshes it tries to reload the CNI (calico by default). This PR updates the `cni.yaml` on the node the enable command runs. If we are in a multinode cluster and one of the other nodes refreshes the cni of the node that refreshes gets reloaded. This means that nodes that do not have the calico interfaces "patched out" will revive calico. We need to either get kube-ovn work next to calico and/or make the patching of kube-ovn a distributed op.

[1] https://github.com/canonical/microk8s/blob/master/microk8s-resources/wrappers/apiservice-kicker#L114

Reply:
Addressed in https://github.com/canonical/microk8s/pull/3307